dataflux_pytorch/dataflux_iterable_dataset.py

```diff
@@ -130,6 +132,11 @@ def __init__(
         self.bucket_name = bucket_name
         self.data_format_fn = data_format_fn
         self.config = config
+        if not self._has_permissions():
+            logging.warning(
+                f"Composed download disabled as permissions to create or delete objects is missing."
```

Comment:
Is there any way to log the specific permission that is missing? This will make it easier for users to debug when they hit this error

Reply:
moved the function outside the class.